app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntryEffect.kt

```diff
@@ -1,10 +1,21 @@
 package org.simple.clinic.bp.entry
 
+import org.threeten.bp.Instant
 import java.util.UUID
 
 sealed class BloodPressureEntryEffect
 
-object PrefillDateForNewEntry : BloodPressureEntryEffect()
+data class PrefillDate(val date: Instant? = null) : BloodPressureEntryEffect() {
```

Comment:
I feel this should be a sealed class, since the date being `null` is just a signal that we should pre-fill the current date instead of a specific date. What do you think?

Reply:
```kotlin
sealed class PrefillDate {
  object PrefillCurrentDate : PrefillDate()
  
  data class PrefillSpecificDate(val instant: Instant): PrefillDate()
}: BloodPressureEntryEffect()
```

Reply:
I understand, however when we are migrating. I prefer focusing on moving to the newer architecture before refactoring the design of the migration itself. These are two separate problems. In most cases refactoring the Mobius classes may not yield optimum results unless the migration is complete.

However, I am adding this to my list of TODOs and will revisit while refactoring these classes once the migration is done (last PR for this feature).

Reply:
Please keep bringing these points so that I can refer to them when I refactor Mobius classes.

Reply:
Sure, my intention was to add it as a suggestion for the final clean up PR after migration anyway.